### PR TITLE
Resolve Eclipse warnings #223

### LINKED ISSUE #223: Resolve Eclipse warnings
These are all tiny issues, so they can probably be done in a single PR:
```
Description	Resource	Path	Location	Type
Class is a raw type. References to generic type Class<T> should be parameterized	TestUtil.java	/addressbook-level4/src/test/java/seedu/address/testutil	line 218	Java Problem
Comparable is a raw type. References to generic type Comparable<T> should be parameterized	UnmodifiableObservableList.java	/addressbook-level4/src/main/java/seedu/address/commons/core	line 96	Java Problem
The static method readConfig(String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 85	Java Problem
The static method readConfig(String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 109	Java Problem
The static method readConfig(String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 116	Java Problem
The static method saveConfig(Config, String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 108	Java Problem
The static method saveConfig(Config, String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 115	Java Problem
The static method saveConfig(Config, String) from the type ConfigUtil should be accessed in a static way	ConfigUtilTest.java	/addressbook-level4/src/test/java/seedu/address/commons/util	line 122	Java Problem
The value of the field IncorrectCommandAttemptedEvent.command is not used	IncorrectCommandAttemptedEvent.java	/addressbook-level4/src/main/java/seedu/address/commons/events/ui	line 11	Java Problem
The value of the field MainWindow.addressBookName is not used	MainWindow.java	/addressbook-level4/src/main/java/seedu/address/ui	line 48	Java Problem
The value of the field MainWindow.commandBox is not used	MainWindow.java	/addressbook-level4/src/main/java/seedu/address/ui	line 41	Java Problem
The value of the field MainWindow.resultDisplay is not used	MainWindow.java	/addressbook-level4/src/main/java/seedu/address/ui	line 39	Java Problem
The value of the field MainWindow.statusBarFooter is not used	MainWindow.java	/addressbook-level4/src/main/java/seedu/address/ui	line 40	Java Problem
The value of the field MainWindow.userPrefs is not used	MainWindow.java	/addressbook-level4/src/main/java/seedu/address/ui	line 43	Java Problem
Type safety: The method getDeclaredMethod(String, Class...) belongs to the raw type Class. References to generic type Class<T> should be parameterized	TestUtil.java	/addressbook-level4/src/test/java/seedu/address/testutil	line 219	Java Problem
Type safety: Unchecked cast from Node to ListView<ReadOnlyPerson>	PersonListPanelHandle.java	/addressbook-level4/src/test/java/guitests/guihandles	line 40	Java Problem
```

### PR DESCRIPTION
**Pull request by edmundmok**
Fixes #223

### CODE REVIEW

**Comment by edmundmok** [comment]
Adding the getters in `MainWindow` to resolve the warnings caused coverage to fall...

**Comment by pyokagan** [comment]
> Adding the getters in MainWindow to resolve the warnings caused coverage to fall...

Perhaps those fields could be removed?

**Comment by damithc** [comment]
Small coverage drops can be ignored if the extra lines are not worth the effort covering.

**Comment by damithc** [comment]
@edmundmok is this ready?

**Comment by edmundmok** [comment]
@damithc Hi Prof, I'm not sure if getters should be implemented here or if we should remove the fields as @pyokagan has mentioned.

**Comment by damithc** [comment]
Let's remove them. We can add them back when there is a use for it.

**Comment by edmundmok** [comment]
Apparently the unused fields in `MainWindow` such as `resultDisplay` and `commandBox` are used in their respective Gui handles so we cannot remove them...
The other changes should be ready though.

**Comment by damithc** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/se-edu/addressbook-level4/227)*

**Comment by pyokagan** [comment]
@edmundmok 
> Apparently the unused fields in MainWindow such as commandBox are used in GuiHandle to retrieve the appropriate node.

Huh? Do you get any test failures when removing them?

So, here's what I did,

1. Rebased this PR.

2. Apply the following patch, which resolves all the Eclipse warnings (with the exception of the serializable exceptions ones):

```diff
diff --git a/src/main/java/seedu/address/commons/events/ui/IncorrectCommandAttemptedEvent.java b/src/main/java/seedu/address/commons/events/ui/IncorrectCommandAttemptedEvent.java
index cb04295..9d732b7 100644
--- a/src/main/java/seedu/address/commons/events/ui/IncorrectCommandAttemptedEvent.java
+++ b/src/main/java/seedu/address/commons/events/ui/IncorrectCommandAttemptedEvent.java
@@ -1,17 +1,13 @@
 package seedu.address.commons.events.ui;
 
 import seedu.address.commons.events.BaseEvent;
-import seedu.address.logic.commands.Command;
 
 /**
  * Indicates an attempt to execute an incorrect command
  */
 public class IncorrectCommandAttemptedEvent extends BaseEvent {
 
-    private Command command;
-
-    public IncorrectCommandAttemptedEvent(Command command) {
-        this.command = command;
+    public IncorrectCommandAttemptedEvent() {
     }
 
     @Override
diff --git a/src/main/java/seedu/address/logic/commands/Command.java b/src/main/java/seedu/address/logic/commands/Command.java
index 7c0ba2f..c5432cb 100644
--- a/src/main/java/seedu/address/logic/commands/Command.java
+++ b/src/main/java/seedu/address/logic/commands/Command.java
@@ -41,6 +41,6 @@ public abstract class Command {
      * Raises an event to indicate an attempt to execute an incorrect command
      */
     protected void indicateAttemptToExecuteIncorrectCommand() {
-        EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent(this));
+        EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent());
     }
 }
diff --git a/src/main/java/seedu/address/ui/MainWindow.java b/src/main/java/seedu/address/ui/MainWindow.java
index 5dc2d36..cd9076b 100644
--- a/src/main/java/seedu/address/ui/MainWindow.java
+++ b/src/main/java/seedu/address/ui/MainWindow.java
@@ -36,17 +36,11 @@ public class MainWindow extends UiPart {
     // Independent Ui parts residing in this Ui container
     private BrowserPanel browserPanel;
     private PersonListPanel personListPanel;
-    private ResultDisplay resultDisplay;
-    private StatusBarFooter statusBarFooter;
-    private CommandBox commandBox;
     private Config config;
-    private UserPrefs userPrefs;
 
     // Handles to elements of this Ui container
     private VBox rootLayout;
 
-    private String addressBookName;
-
     @FXML
     private AnchorPane browserPlaceholder;
 
@@ -91,9 +85,7 @@ public class MainWindow extends UiPart {
         // Set dependencies
         this.primaryStage = primaryStage;
         this.logic = logic;
-        this.addressBookName = addressBookName;
         this.config = config;
-        this.userPrefs = prefs;
 
         // Configure the UI
         setTitle(appTitle);
@@ -143,9 +135,9 @@ public class MainWindow extends UiPart {
     void fillInnerParts() {
         browserPanel = BrowserPanel.load(browserPlaceholder);
         personListPanel = PersonListPanel.load(getPersonListPlaceholder(), logic.getFilteredPersonList());
-        resultDisplay = ResultDisplay.load(getResultDisplayPlaceholder());
-        statusBarFooter = StatusBarFooter.load(getStatusbarPlaceholder(), config.getAddressBookFilePath());
-        commandBox = CommandBox.load(getCommandBoxPlaceholder(), logic);
+        ResultDisplay.load(getResultDisplayPlaceholder());
+        StatusBarFooter.load(getStatusbarPlaceholder(), config.getAddressBookFilePath());
+        CommandBox.load(getCommandBoxPlaceholder(), logic);
     }
 
     private AnchorPane getCommandBoxPlaceholder() {
diff --git a/src/test/java/guitests/guihandles/PersonListPanelHandle.java b/src/test/java/guitests/guihandles/PersonListPanelHandle.java
index 528e99f..c3fb889 100644
--- a/src/test/java/guitests/guihandles/PersonListPanelHandle.java
+++ b/src/test/java/guitests/guihandles/PersonListPanelHandle.java
@@ -36,7 +36,6 @@ public class PersonListPanelHandle extends GuiHandle {
         return personList.getSelectionModel().getSelectedItems();
     }
 
-    @SuppressWarnings("unchecked")
     public ListView<ReadOnlyPerson> getListView() {
         return getNode(PERSON_LIST_VIEW_ID);
     }
```

3. Run `xvfb-run gradle check`

And I get zero test failures.

**Comment by damithc** [comment]
Regarding unused variables, I think we put those local variable to represent the composition structure of the GUI. i.e. parent/container component holds a pointer to child/sub components. The idea was that the parent components may need to pass information down to sub-components. However, the event driven approach makes such information paths unnecessary because any component can subscribe to an event without waiting for the parent component to pass the information to it. As a result, some of those variables are not used.  We can keep them if we want the conceptual hierarchy of components to be reflected in code as composition links. But I'm ok to remove them too.

**Comment by edmundmok** [comment]
@pyokagan Oh dear, I think I must have removed the loading of `ResultDisplay` etc, in my attempt to get rid of the error without realizing it, causing the errors. You are right there won't be any errors.

**Comment by pyokagan** [comment]
@damithc 
> We can keep them if we want the conceptual hierarchy of components to be reflected in code as composition links. But I'm ok to remove them too.

I agree that that's a good design that the code should aspire to. However, the UiPart architecture, as it is implemented currently, makes that hard :-P #237 implements some steps towards resolving that (it adds a `getRoot()` method to `UiPart` which retrieves the root of the scene graph), but after that is merged, some more work will still need to be done (rewiring the code such that callers themselves add the UiPart to the scene graph rather than letting the UiPart constructor do it). Once that is all done, we shouldn't have these "unused variables" problems any more.

Still, I think we should remove them for now because having too many warnings cluttering up the warnings list makes it hard to spot *new* warnings.

**Comment by damithc** [comment]
> Still, I think we should remove them for now because having too many warnings cluttering up the warnings list makes it hard to spot new warnings.

👍

**Comment by edmundmok** [comment]
Updated.

File: src/test/java/guitests/guihandles/PersonListPanelHandle.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public PersonListPanelHandle(GuiRobot guiRobot, Stage primaryStage) {
         return personList.getSelectionModel().getSelectedItems();
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (pyokagan) [inline src/test/java/guitests/guihandles/PersonListPanelHandle.java]: Remove this.

Comment (edmundmok) [inline src/test/java/guitests/guihandles/PersonListPanelHandle.java]: Removed.

**Pull request merged**